File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    removeDuplicates(nums) {
        let end_idx = nums.length - 1;
        let curr_idx = 0;

        //will reach deleted if there is at least one pair
        //will reach end of array if there are no pairs (all elems are unique so none are deleted)
        while (curr_idx != end_idx) {
            if (nums[curr_idx] == nums[curr_idx + 1]) {
                this.remove_elem(nums, curr_idx + 1, end_idx);
                //update end idx since we deleted an elem
                end_idx -= 1;
            } else {
                //if the next element is unique
                curr_idx += 1; 
            }
        }

        //number of elems up to end idx
        return end_idx + 1;
    }

    remove_elem(nums, idx, end_idx) {
        //remove element at a given index
        for (let i = idx; i < end_idx; i++) {
            nums[i] = nums[i+1];
        }
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-3.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        r = len(nums) - 1
        l = 0
        while l <= r:
            p = (l + r) // 2
            if nums[p] == target:
                return p
            if target > nums[p]:
                l = p + 1
            else:
                r = p - 1
        return l